great_expectations/datasource/datasource.py

```diff
@@ -173,12 +173,17 @@ def save_config(self):
         Returns:
             None
         """
-        if self._data_context is not None:
-            self._data_context._save_project_config()
-        else:
-            config_filepath = "great_expectations.yml"
-            with open(config_filepath, 'w') as config_file:
-                yaml.dump(self._datasource_config, config_file)
+
+        #FIXME: Not clear if deprecating this causes any breakages elsewhere...
```

Comment:
Context? What's going on in this change?

Reply:
Good catch.

The immediate issue was that `DataSource` reaching into the `DataContext`s internal `_save_config` method was failing for `ConfigOnlyDataContext`---which doesn't have a `_save_config` method.

The broader issue, which remains unresolved (!!!), is that `DataContext`s are now INTENDED to use  their `project_configs` as an unambiguous source of truth. I believe that this is a better pattern than using the config and the yml in tandem and hoping they'll stay in sync.

But
1. it's not yet fully documented
2. there may be additional checks or convenience methods required for DataContext to live up to that contract, and
3. we need to do a thorough audit to make sure no other classes are using now-deprecated methods to read/update the `DataContext._project_config`

Reply:
I think this makes a lot of sense, but it seems like the datasource still needs to be able to signal to the DataContext that its config has changed (e.g. when a generator is changed via access to the datasource)